### ISSUE TITLE
active_model_serializers gem compatability?

### ISSUE THREAD

**Issue by followfung** [ohler55/oj#138]
Is there a way to get Oj to work with the active_model_serializers gem? Or is the way I'm trying to use this gem wrong? (Working on a Rails 3 project)

Currently I am doing this in my controllers:

```
render json: Oj.dump(@objects, mode: :compat)
```

but it seems to ignore what I have in my serializers.

**Comment by erlandsona** [comment]
I was literally about to ask this question! Thank you so much @followfung!

**Comment by followfung** [comment]
@erlandsona The code snippet above is relevant to a Rails 3 application that I was working on. It's been a long time since I've worked on the project. Unfortunately I don't have access to the source code anymore (code is owned by my previous workplace) so I may not be able to help you much.

How is your project set up?

**Comment by erlandsona** [comment]
@followfung are you on slack? Either way, I'm running a Rails 4.2.4 app on Ruby 2.2.2 with Puma. I'm using AMS 0.10.0 which I know is still unstable but it's the only version that's gotten me even close to working with OJ version 2.13.0 and I'm using the oj_mimic_json thing???

**Comment by followfung** [comment]
Sorry for the late reply,

If I remember right, I was working with active_model_serializers 0.8.x. 

The snippet that I posted is pretty old, and I doubt it would work with the current RC of active_model_serializers as it introduces breaking changes (not backwards compatible with 0.8.x). 

My implementation was essentially a hack to begin with to get Oj to play nice with active_model_serializers . I only found that `target.active_model_serializer.new(target, options)` would work in my case by digging through the source code of the active_model_serializers gem.

I'm not sure what changes in Rails4 could break this snippet either.

With that in mind, what my code snippet did was introduce the `json_for` helper method. 
It takes my `@posts` object (ActiveRecord) and force it to go through active_model_serializers , and then I pass the serialized object to `Oj.dump()`.

From what I can see, whatever changes were made removed the ability to invoke `target.active_model_serializer.new(target, options)`.

I hope this helps
